docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -285,6 +285,11 @@ Documentation fragments
 
 If you're writing multiple related modules, they may share common documentation, such as authentication details, file mode settings, ``notes:`` or ``seealso:`` entries. Rather than duplicate that information in each module's ``DOCUMENTATION`` block, you can save it once as a doc_fragment plugin and use it in each module's documentation. In Ansible, shared documentation fragments are contained in a ``ModuleDocFragment`` class in `lib/ansible/plugins/doc_fragments/ <https://github.com/ansible/ansible/tree/devel/lib/ansible/plugins/doc_fragments>`_. To include a documentation fragment, add ``extends_documentation_fragment: FRAGMENT_NAME`` in your module's documentation.
 
+Modules should only use a doc_fragment if it will implement all of the interface documented there in
+a manner that behaves the same as the existing modules which implement that fragment.  The goal is
+that any parameter listed in doc_fragments will behave identically when used in another module
+implementing the doc_fragment.
```

Comment:
not `all`, modules can use partial refs to doc fragments if they are setup that way.

i.e 
```extends_documentation.fragment: microsoft.azure.documentation.azure_auth```